scenarios/throughput_stress.go

```diff
@@ -183,43 +180,30 @@ func (t *tpsExecutor) Run(ctx context.Context, info loadgen.ScenarioInfo) error
 	}
 	t.runID = info.RunID
 
-	// Track start time of current run
-	currentRunStartTime := time.Now()
-
-	// Add search attribute, if it doesn't exist yet, to query for workflows by run ID.
-	// Running this on resume, too, in case a previous Omes run crashed before it could add the search attribute.
-	if err := loadgen.InitSearchAttribute(ctx, info, ThroughputStressScenarioIdSearchAttribute); err != nil {
-		return err
-	}
-
 	t.lock.Lock()
 	isResuming := t.isResuming
 	currentState := *t.state
 	t.lock.Unlock()
 
 	if isResuming {
 		info.Logger.Info(fmt.Sprintf("Resuming scenario from state: %#v", currentState))
-		info.Configuration.StartFromIteration = int(currentState.CompletedIterations) + 1
```

Comment:
This was technically a good idea, but easily causes one-off issues. For example, when the all iterations were created but the run crashes and resumes, it would error since the start iteration cannot be larger than the target number of iterations. And now with the `WorkflowExecutionAlreadyStarted` handling, this is safe.